requestmanager/requestmanager_test.go

```diff
@@ -1027,22 +1026,22 @@ func (fph *fakePeerHandler) AllocateAndBuildMessage(p peer.ID, blkSize uint64,
 	}
 }
 
-func readNNetworkRequests(ctx context.Context,
-	t *testing.T,
-	requestRecordChan <-chan requestRecord,
-	count int) []requestRecord {
-	requestRecords := make([]requestRecord, 0, count)
+func readNNetworkRequests(ctx context.Context, t *testing.T, td *testData, count int) []requestRecord {
```

Comment:
oh the simplied sorting in this method is a sad state of affairs.... your correction works, but I wonder how I ended writing the tests this way to count on sorting order